### PR TITLE
[ci-visibility] Support `mocha` test suites

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
Add visibility for test session and test suite in `mocha`.

### Motivation
Expand visibility from just tests to suites and test processes. 

### Plugin Checklist

- [x] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/dd-trace-js/blob/master/docs/API.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

**Review by rochdev — CHANGES_REQUESTED** [review]
The PR looks good for the most part, but there are several event listeners that are not explicitly bound to an `AsyncResource`. This could mean disconnected traces or spans that would never finish in some case. Any time an event is published on a channel it should always be bound to ensure it will be in the right scope regardless of where it was emitted.

**Comment by juan-fernandez** [comment]
> The PR looks good for the most part, but there are several event listeners that are not explicitly bound to an `AsyncResource`. This could mean disconnected traces or spans that would never finish in some case. Any time an event is published on a channel it should always be bound to ensure it will be in the right scope regardless of where it was emitted.

good catch. That should be fixed in https://github.com/DataDog/dd-trace-js/pull/2172/commits/dfcda0ae3e63c2bcaf3ead80fe5cdc838ca39302

**Review by rochdev — APPROVED** [review]

**Comment by juan-fernandez** [comment]
let's hold off on merging this for a bit until we figure out a potential issue in the intake

File: packages/dd-trace/src/encode/agentless-ci-visibility.js [hunk 1]
```diff
@@ -57,22 +130,38 @@ class AgentlessCiVisibilityEncoder extends AgentEncoder {
     this._encodeNumber(bytes, content.start)
     this._encodeString(bytes, 'duration')
     this._encodeNumber(bytes, content.duration)
+    // only for v2
```
Comment (juan-fernandez) [inline packages/dd-trace/src/encode/agentless-ci-visibility.js:133]: Add `TODO` here: explain why this is needed and that it needs a solution

**Comment by juan-fernandez** [comment]
added the TODO comment. Now solving the conflict

**Review by rochdev — APPROVED** [review]

**Pull request merged**